frontends/p4/parseAnnotations.cpp

```diff
@@ -32,10 +32,11 @@ ParseAnnotations::HandlerMap ParseAnnotations::standardHandlers() {
         PARSE_EMPTY("unroll"_cs),
         PARSE_EMPTY("nounroll"_cs),
 
-        // string literal argument.
-        PARSE(IR::Annotation::nameAnnotation, StringLiteral),
-        PARSE(IR::Annotation::deprecatedAnnotation, StringLiteral),
-        PARSE(IR::Annotation::noWarnAnnotation, StringLiteral),
+        // String arguments. These are allowed to contain concatenation which will be
```

Comment:
since now it is parsed as a whole expression, what happens with other binary operators? E.g. what is produced with `"foo" | "bar"` and so on?

Reply:
You would get a normal error message:

```p4
spec-issue1297-string-cat-err-3-anno.p4(3): [--Werror=type-error] error: |: cannot be applied to expression '"a"' with type 'string'
@name("a" | "b")
        ^
```

But the problem is when you do something like `@name(4)`, let me check that.

Reply:
OK, now the error message looks like this:

```p4
spec-issue1297-string-cat-err-2-anno.p4(2): [--Werror=type-error] error: 4: @name annotation's value must be a string
@name(4)
      ^
```

I've added a validation pass similar to what we do for the `@match`. In this case, however, the correct working of this pass depends on it running after constant folding so that `++` is resolved.